fe/fe-core/src/main/java/com/starrocks/catalog/TabletInvertedIndex.java

```diff
@@ -311,6 +305,136 @@ public void tabletReport(long backendId, Map<Long, TTablet> backendTablets,
+                    continue;
+                }
+
+                // validate database
+                long dbId = tabletMeta.getDbId();
+                Database database = localMetastore.getDb(dbId);
+                if (database == null) {
+                    database = recycleBin.getDatabase(dbId);
+                    if (database != null) {
+                        isInRecycleBin = true;
+                    } else {
+                        deleteTabletByConsistencyChecker(tabletId, backendId,
+                                "database " + dbId + " doesn't exist", invalidTablets);
+                        continue;
+                    }
+                }
+
+                try {
+                    database.readLock();
+
+                    // validate table
+                    long tableId = tabletMeta.getTableId();
+                    com.starrocks.catalog.Table table = database.getTable(tableId);
+                    if (table == null) {
+                        table = recycleBin.getTable(dbId, tableId);
+                        if (table != null) {
+                            isInRecycleBin = true;
+                        } else {
+                            deleteTabletByConsistencyChecker(tabletId, backendId,
+                                    "table " + dbId + "." + tableId + " doesn't exist", invalidTablets);
+                            continue;
+                        }
+                    }
+
+                    // validate partition
+                    long partitionId = tabletMeta.getPartitionId();
+                    Partition partition = table.getPartition(partitionId);
+                    if (partition == null) {
+                        partition = recycleBin.getPartition(partitionId);
+                        if (partition != null) {
+                            isInRecycleBin = true;
+                        } else {
+                            deleteTabletByConsistencyChecker(tabletId, backendId,
+                                    "partition " + dbId + "." + tableId + "." + partitionId + " doesn't exist",
+                                    invalidTablets);
+                            continue;
+                        }
+                    }
+
+                    // validate index
+                    long indexId = tabletMeta.getIndexId();
+                    MaterializedIndex index = partition.getIndex(indexId);
+                    if (index == null) {
+                        deleteTabletByConsistencyChecker(tabletId, backendId,
+                                "materialized index " + dbId + "." + tableId + "." +
+                                        partitionId + "." + indexId + " doesn't exist",
+                                invalidTablets);
+                        continue;
+                    }
+
+                    if (!table.isCloudNativeTableOrMaterializedView()) {
```

Comment:
Why not check tablet for MaterializedView?

Reply:
it's actually cloud table and cloud mv, you can see the detail impl :-)